src/jdbc/cubrid/jdbc/net/BrokerHandler.java

```diff
@@ -227,6 +228,20 @@ public static int statusBroker(String ip, int port, int process, byte[] session,
         return status;
     }
 
+    public static boolean isValid(String ip, int port, int process, byte[] session, int timeout)
+        throws SQLException {
+
+        try {
+            if (BrokerHandler.statusBroker(ip, port, process, session, timeout) == -2) {
+                return false;
+            }
+        } catch (Exception e) {
+            throw new SQLException("Invalid Connection");
+        }
```

Comment:
According to JDBC Spec, SQLException should be thrown **if the value supplied for timeout is less then 0**.